core/workflow/loadDPL.go

```diff
@@ -65,8 +65,27 @@ func LoadDPL(tasks []*task.Class, rootRoleName string) (workflow Role, err error
 	return workflow, nil
 }
 
-// RoleToYAML exists to avoid exporting aggregatorRole. Simply put, this function is nothing but a wrapper
-// which allows marshalling Roles outside the workflow package
+// RoleToYAML exists to avoid exporting aggregatorRole/iteratorRole.
+// Simply put, this function is nothing but a wrapper which allows marshalling Roles outside the workflow package
 func RoleToYAML(input Role) ([]byte, error) {
-	return yaml.Marshal(input.(*aggregatorRole))
+	switch input.(type) {
+	case *aggregatorRole:
+		return yaml.Marshal(input.(*aggregatorRole))
+	case *iteratorRole:
+		return yaml.Marshal(input.(*iteratorRole))
+	default:
+		return nil, nil
+	}
+}
+
+func LoadWorkflow(input []byte) (workflow Role, err error) {
+	root := new(aggregatorRole)
+	err = yaml.Unmarshal(input, &root)
+	if err != nil {
+		return workflow, err
+	}
+
+	workflow = root
+
+	return workflow, nil
 }
```

Comment:
At this point, I do not feel that `loadDPL.go` should be a separate file. IMO, `load.go` is a more suitable location for it's functions. Let me know if it's alright to move its contents.

Reply:
Feel free to make this change in a new PR